addon/model/operations/operation-algorithms.ts

```diff
@@ -7,13 +7,182 @@ import RangeMapper, {
 } from '@lblod/ember-rdfa-editor/model/range-mapper';
 import { RelativePosition } from '@lblod/ember-rdfa-editor/model/util/types';
 import ModelText from '@lblod/ember-rdfa-editor/model/model-text';
+import ModelElement from '../model-element';
 
 export type OperationAlgorithmResponse<T> = { mapper: RangeMapper } & T;
 /**
  * A shared library of algorithms to be used by operations only
  * Any use outside of operations is not supported
  */
+
 export default class OperationAlgorithms {
+  static removeNew(
+    range: ModelRange
+  ): OperationAlgorithmResponse<{ removedNodes: ModelNode[] }> {
+    //config consts
+    const cantMergeIntoTypes = ['a'];
+    const cantRemoveOpeningTagNodeTypes = [
+      'a',
+      'ul',
+      'li',
+      'td',
+      'tr',
+      'table',
+    ];
+
+    //start algorithm
+    if (range.collapsed) {
+      return {
+        removedNodes: [],
+        mapper: new RangeMapper([buildPositionMapping(range, range.start)]),
+      };
+    }
+
+    range.normalize();
+
+    //split end and start if they are inside a text node
+    if (range.start.isInsideText()) {
+      range.start.split();
+    }
+    if (range.end.isInsideText()) {
+      range.end.split(true);
+    }
+
+    //get nodes that will move after the delition operation
+    //these are siblings of the closest node to the end of the range (if any)
+    //assumption: nodeAfter doesn't grab anything if the next node to the end of the range is one level up
+    const nodesToMove: ModelNode[] = [];
+    let nextNode = range.end.nodeAfter();
+    while (nextNode) {
+      nodesToMove.push(nextNode);
+      nextNode = nextNode.nextSibling;
+    }
+
+    //grab all nodes inside the range
+    //assumption: the only partial nodes that treewalker grabs are the ones that have opening tags in the selection
+    //assumption: opening tag nodes are always parents of the last node in range
+    const allNodes: ModelNode[] = [];
+    const walker = new ModelTreeWalker({ range: range });
+    for (const node of walker) {
+      allNodes.push(node);
+    }
+
+    //get all nodes that are fully contained in the range
+    //ie [<span><text>abc</text>]</span>
+    //would grab just the text node
+    const confinedNodes: ModelNode[] = [];
```

Comment:
see prev comment about unpacking iterables

Reply:
not sure that i can simplify this